### PR TITLE
push changes only when there are changes

### PR DESCRIPTION
**Pull request by hilary**
Fix a problem with #31 
- when merging to main, the branch already has the bundled spec: the subsequent push fails because there are no changes
- in any case, no need to push when the bundled spec hasn't changed! Now checking to see if it has actually changed before staging, committing and pushing
- using `git status --porcelain` instead of plumbing commands because the check reports as a failure if any step does something which returns a non-zero exit code. Note: it might be possible to redirect the exit code to the void and survive, but that defeats the purpose of using a git plumbing command. Anyhoo...

### CODE REVIEW

File: .github/workflows/cd.yml [hunk 1]
```diff
@@ -4,7 +4,7 @@ on: workflow_dispatch
 
 jobs:
   build:
-    runs-on: ubuntu-20.04
+    runs-on: ubuntu-latest
```
Comment (lbirdeau) [inline .github/workflows/cd.yml:7]: why the change to 'latest'?

**Review by lbirdeau — APPROVED** [review]
LGTM

File: .github/workflows/cd.yml (hunk 1 above)
Comment (hilary) [reply .github/workflows/cd.yml:7]: The specific `ubuntu-20.04` tag turns out to be listed as `experimental`, whatever that means.
